src/XMakeBuildEngine/Definition/ToolsetReader.cs

```diff
@@ -471,6 +486,113 @@ bool accumulateProperties
+        /// </summary>
+        private static void AppendStandardProperties(
+            PropertyDictionary<ProjectPropertyInstance> properties,
+            PropertyDictionary<ProjectPropertyInstance> globalProperties,
+            string version,
+            string root,
+            string toolsPath)
+        {
+            if (NativeMethodsShared.IsMono)
+            {
+                var v4Dir = FrameworkLocationHelper.GetPathToDotNetFrameworkV40(DotNetFrameworkArchitecture.Current)
+                            + Path.DirectorySeparatorChar;
+                var v35Dir = FrameworkLocationHelper.GetPathToDotNetFrameworkV35(DotNetFrameworkArchitecture.Current)
+                             + Path.DirectorySeparatorChar;
+
+                if (root == null)
+                {
+                    var libraryPath = NativeMethodsShared.FrameworkBasePath;
+                    if (toolsPath.StartsWith(libraryPath))
+                    {
+                        root = Path.GetDirectoryName(toolsPath);
+                        if (toolsPath.EndsWith("bin"))
+                        {
+                            root = Path.GetDirectoryName(root);
+                        }
+                    }
+                    else
+                    {
+                        root = libraryPath;
+                    }
+                }
+
+                root += Path.DirectorySeparatorChar;
+
+                // Global properties cannot be overwritten
+                if (globalProperties["FrameworkSDKRoot"] == null && properties["FrameworkSDKRoot"] == null)
+                {
+                    properties.Set(ProjectPropertyInstance.Create("FrameworkSDKRoot", root, true, false));
+                }
+                if (globalProperties["MSBuildToolsRoot"] == null && properties["MSBuildToolsRoot"] == null)
+                {
+                    properties.Set(ProjectPropertyInstance.Create("MSBuildToolsRoot", root, true, false));
+                }
+                if (globalProperties["FrameworkSDKRoot"] == null && properties["FrameworkSDKRoot"] == null)
+                {
+                    properties.Set(ProjectPropertyInstance.Create("FrameworkSDKRoot", root, true, false));
+                }
+                if (globalProperties["MSBuildFrameworkToolsPath"] == null
+                    && properties["MSBuildFrameworkToolsPath"] == null)
+                {
+                    properties.Set(ProjectPropertyInstance.Create("MSBuildFrameworkToolsPath", toolsPath, true, false));
+                }
+                if (globalProperties["MSBuildFrameworkToolsPath32"] == null
+                    && properties["MSBuildFrameworkToolsPath32"] == null)
+                {
+                    properties.Set(
+                        ProjectPropertyInstance.Create("MSBuildFrameworkToolsPath32", toolsPath, true, false));
+                }
+                /*if (globalProperties["MSBuildToolsPath"] == null && properties["MSBuildToolsPath"] == null)
```

Comment:
nit: please don't check in commented out code (at least not without a description of why)